File: sdk/js/src/models/ChatEndpointHealth.js
/**
 * ChatEndpointHealth class representing health status for a chat endpoint.
 */
export default class ChatEndpointHealth {
  /**
   * @param {Object} health - Information about the endpoint health.
   * @param {string} [health.EndpointGUID] - Globally unique identifier for the chat endpoint.
   * @param {string} [health.TenantGUID] - Globally unique identifier for the tenant.
   * @param {string|null} [health.Name=null] - Name of the chat endpoint.
   * @param {string} [health.EndpointType='Completion'] - Endpoint type: Embedding or Completion.
   * @param {boolean} [health.Monitored=false] - Indicates whether the endpoint is monitored (default is false).
   * @param {boolean|null} [health.Healthy=null] - Health status; null when not yet checked (default is null).
   * @param {Date|string|null} [health.LastCheckedUtc=null] - Last check timestamp in UTC (default is null).
   * @param {string|null} [health.LastError=null] - Last error message, if any (default is null).
   * @param {number} [health.ConsecutiveSuccesses=0] - Consecutive successful checks (default is 0).
   * @param {number} [health.ConsecutiveFailures=0] - Consecutive failed checks (default is 0).
   * @param {number|null} [health.UptimePercentage=null] - Uptime percentage over the check history (default is null).
   * @param {Array<Object>} [health.CheckHistory=[]] - Recent health check samples ({ TimestampUtc, Success, DurationMs }).
   */
  constructor(health = {}) {
    const {
      EndpointGUID = null,
      TenantGUID = null,
      Name = null,
      EndpointType = 'Completion',
      Monitored = false,
      Healthy = null,
      LastCheckedUtc = null,
      LastError = null,
      ConsecutiveSuccesses = 0,
      ConsecutiveFailures = 0,
      UptimePercentage = null,
      CheckHistory = [],
    } = health;

    this.EndpointGUID = EndpointGUID; // Unique identifier for the chat endpoint
    this.TenantGUID = TenantGUID; // Unique identifier for the tenant
    this.Name = Name; // Name of the chat endpoint
    this.EndpointType = EndpointType; // Endpoint type (Embedding or Completion)
    this.Monitored = Monitored; // Indicates if the endpoint is monitored
    this.Healthy = Healthy; // Health status (null when not yet checked)
    this.LastCheckedUtc = LastCheckedUtc ? new Date(LastCheckedUtc) : null; // Last check timestamp
    this.LastError = LastError; // Last error message
    this.ConsecutiveSuccesses = ConsecutiveSuccesses; // Consecutive successful checks
    this.ConsecutiveFailures = ConsecutiveFailures; // Consecutive failed checks
    this.UptimePercentage = UptimePercentage; // Uptime percentage
    this.CheckHistory = CheckHistory; // Recent health check samples
  }
}

File: src/LiteGraph.Server/Classes/ChatEndpointHealth.cs
namespace LiteGraph.Server.Classes
{
    using System;
    using System.Collections.Generic;

    /// <summary>
    /// Chat endpoint health status.
    /// </summary>
    public class ChatEndpointHealth
    {
        #region Public-Members

        /// <summary>
        /// Endpoint GUID.
        /// </summary>
        public Guid EndpointGUID { get; set; } = Guid.NewGuid();

        /// <summary>
        /// Tenant GUID.
        /// </summary>
        public Guid TenantGUID { get; set; } = Guid.NewGuid();

        /// <summary>
        /// Endpoint name.
        /// </summary>
        public string Name { get; set; } = null;

        /// <summary>
        /// Endpoint type.
        /// </summary>
        public ChatEndpointTypeEnum EndpointType { get; set; } = ChatEndpointTypeEnum.Completion;

        /// <summary>
        /// Whether the endpoint is being monitored.  False when health checks are disabled or the endpoint is inactive.
        /// </summary>
        public bool Monitored { get; set; } = false;

        /// <summary>
        /// Whether the endpoint is healthy.  Null while monitoring has not yet reached a verdict.
        /// </summary>
        public bool? Healthy { get; set; } = null;

        /// <summary>
        /// Timestamp of the most recent probe, in UTC.  Null when no probe has run.
        /// </summary>
        public DateTime? LastCheckedUtc { get; set; } = null;

        /// <summary>
        /// Most recent probe error.  Null when the last probe succeeded.
        /// </summary>
        public string LastError { get; set; } = null;

        /// <summary>
        /// Consecutive successful probes.
        /// </summary>
        public int ConsecutiveSuccesses { get; set; } = 0;

        /// <summary>
        /// Consecutive failed probes.
        /// </summary>
        public int ConsecutiveFailures { get; set; } = 0;

        /// <summary>
        /// Percentage of successful probes over the retained history window.  Null when no probes have run.
        /// </summary>
        public double? UptimePercentage { get; set; } = null;

        /// <summary>
        /// Rolling probe history, oldest first.  Retained for 24 hours.
        /// </summary>
        public List<ChatEndpointHealthSample> CheckHistory { get; set; } = new List<ChatEndpointHealthSample>();

        #endregion

        #region Private-Members

        #endregion

        #region Constructors-and-Factories

        /// <summary>
        /// Instantiate.
        /// </summary>
        public ChatEndpointHealth()
        {

        }

        #endregion

        #region Public-Methods

        #endregion

        #region Private-Methods

        #endregion
    }
}
